### PR TITLE
Governance Action Scripts

### LINKED ISSUE #225: Governance Operations: Collateral Add
Depends on #241 
- List likely governance operations as Primary Governance Operations
- Create scripts for all PGOs
- Demonstrate success of scripts on live testnet

### LINKED ISSUE #241: Governance Operations: Update NFTRenderer
Use governance actions to update the NFTRenderer without new protocol deployment

### PR DESCRIPTION
**Pull request by 0xdavinchee**
Closes #225 
Closes #241 

TODOS
- [x] Update README on how to select a profile
- [x] Update README on the different governance scripts and how to run them
- [x] Add unit tests for Propose Update NFT Renderer

## Scripts
- Propose Add Collateral Script
- Propose Update NFT Renderer Script
- General Queue Proposal Script
- General Execute Proposal Script
- `JSONScript.s.sol` which helps with building JSON, parsing JSON, and parsing the proposal ID

## Tests
- Fix existing AddCollateral tests: favor `encodeWithSelector` over `encodeWithSignature`.
  - Root issue was that structs must be declared as tuples

## Misc
- New governance profile in `foundry.toml` with more `fs_permissions`
- `deploy:anvil` in `package.json` handles the additional steps of copying broadcast file + node parse..

### CODE REVIEW

File: script/gov/UpdateNFTRendererAction/ProposeUpdateNFTRenderer.s.sol [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      vm.serializeAddress(objectKey, 'nftRendererAddress', nftRendererAddress);
+      string memory jsonOutput =
+        _buildProposalParamsJSON(proposalId, objectKey, targets, values, calldatas, description, descriptionHash);
+      vm.writeJson(jsonOutput, string.concat('./gov-output/', stringProposalId, '-update-nft-renderer-proposal.json'));
+    }
+
+    // Expected JSON output:
+    // {
+    //   "proposalId": uint256,
+    //   "nftRendererAddress": address,
```
Comment (pi0neerpat) [inline script/gov/UpdateNFTRendererAction/ProposeUpdateNFTRenderer.s.sol:71]: Same question as above RE: will this be included in the output file

**Review by daopunk — APPROVED** [review]
Making a separate issue for gov simulation, but approving this on condition that unit tests are included in the PR.

File: script/gov/AddCollateralAction/ProposeAddCollateral.s.sol [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    assert(proposalId == gov.hashProposal(targets, values, calldatas, descriptionHash));
+
+    {
+      // Build the JSON output
+      string memory objectKey = 'PROPOSE_ADD_COLLATERAL_KEY';
+      vm.serializeString(objectKey, 'newCollateralAddress', stringCAddress);
+      vm.serializeString(objectKey, 'newCollateralType', stringCType);
+      string memory jsonOutput =
+        _buildProposalParamsJSON(proposalId, objectKey, targets, values, calldatas, description, descriptionHash);
+      vm.writeJson(jsonOutput, string.concat('./gov-output/', stringProposalId, '-add-collateral-proposal.json'));
```
[the remark this answers is not in the document]
Comment (0xdavinchee) [reply script/gov/AddCollateralAction/ProposeAddCollateral.s.sol:88]: we have two options:
1. create the folder and save to git (more ideal as this folder will likely be created once and that's it)
2. have a check if folder exists, if it doesn't, we can create it on the spot

File: script/gov/AddCollateralAction/ProposeAddCollateral.s.sol [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    }
+
+    // Expected JSON output:
+    // {
+    //   "proposalId": uint256,
+    //   "newCollateralAddress": string,
+    //   "newCollateralType": bytes32,
```
[the remark this answers is not in the document]
Comment (0xdavinchee) [reply script/gov/AddCollateralAction/ProposeAddCollateral.s.sol:95]: yes, while it is not necessary for the queue/execution, it is some extra information serialized to the JSON object in Foundry before being written to the actual file.

File: script/gov/UpdateNFTRendererAction/ProposeUpdateNFTRenderer.s.sol (hunk 1 above)
Comment (0xdavinchee) [reply script/gov/UpdateNFTRendererAction/ProposeUpdateNFTRenderer.s.sol:71]: yes

File: script/gov/AddCollateralAction/ProposeAddCollateral.s.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (0xdavinchee) [reply script/gov/AddCollateralAction/ProposeAddCollateral.s.sol:88]: I went with option 1 and created a folder with a README explaining what the files in there are used for.

**Review by pi0neerpat — APPROVED** [review]

**Pull request merged**